nodeup/pkg/model/bootstrap_client.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package model
+
+import (
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/awsup"
+	"k8s.io/kops/upup/pkg/fi/nodeup/nodetasks"
+)
+
+// SecretBuilder writes secrets
+type BootstrapClientBuilder struct {
+	*NodeupModelContext
+}
+
+func (b BootstrapClientBuilder) Build(c *fi.ModelBuilderContext) error {
+	if b.IsMaster || !b.UseKopsControllerForNodeBootstrap() {
+		return nil
+	}
+
+	var authenticator fi.Authenticator
+	var err error
+	switch kops.CloudProviderID(b.Cluster.Spec.CloudProvider) {
+	case kops.CloudProviderAWS:
+		authenticator, err = awsup.NewAWSAuthenticator()
+	}
```

Comment:
Probably should check `default` in the switch or otherwise handle the `authenticator == nil` case

Reply:
fixed